src/scenes/Activities/Favorite/store.js

```diff
@@ -0,0 +1,92 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import _ from 'lodash';
+import { Routes, SortBy, OracleStatus } from 'constants';
+import { queryAllTopics, queryAllOracles } from '../../../network/graphql/queries';
+import Topic from '../../../stores/models/Topic';
+import Oracle from '../../../stores/models/Oracle';
+
+const INIT_VALUES_FAVPAGE = {
+  loaded: false, // loading state?
+  loadingMore: true, // for loading icon?
+  displayList: [], // data list
+};
+
+export default class {
+  @observable favList = JSON.parse(localStorage.getItem('bodhi_dapp_favList')) || []; // Data example: '21e389b909c7ab977088c8d43802d459b0eb521a'
+
+  @observable loaded = INIT_VALUES_FAVPAGE.loaded
+  @observable displayList = INIT_VALUES_FAVPAGE.displayList
+  @observable loadingMore = INIT_VALUES_FAVPAGE.loadingMore
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.app.wallet.addresses + this.app.global.syncBlockNum,
+      () => {
+        if (this.app.ui.location === Routes.FAVORITE) {
+          this.init();
+        }
+      }
+    );
+    reaction(
+      () => this.displayList,
+      () => {
+      }
+    );
+    reaction(
+      () => this.favList,
+      async () => {
+        this.displayList = await this.fetchFav();
+      }
+    );
+  }
+
+  updateLocalStorageFavList() {
+    localStorage.setItem('bodhi_dapp_favList', JSON.stringify(this.favList));
+  }
+
+  @action
+  setFavorite = (topicAddress) => {
+    if (this.isInFavorite(topicAddress)) this.favList = this.favList.filter(x => x !== topicAddress);
+    else {
+      this.favList.push(topicAddress);
+      this.favList.replace(this.favList);
```

Comment:
why do you need replace? `this.favList.push(topicAddress)` should handle any refreshes needed since its observable.